plugins/ci/pkg/models/models.go

```diff
@@ -46,18 +46,24 @@ type HelmConfig struct {
 	Path       string                 `yaml:"path"`
 	Repo       string                 `yaml:"repo"`
 	Chart      string                 `yaml:"chart"`
+	FluxFile   string                 `yaml:"fluxFile"`
+	Version    string                 `yaml:"version"`
 	ValuesFile string                 `yaml:"valuesFile"`
 	Values     map[string]interface{} `yaml:"values"`
 }
 
 func (hc *HelmConfig) IsRemote() bool {
-	return hc.Repo != ""
+	return hc.Repo != "" && !hc.IsFluxFile()
```

Comment:
IIUC all the FluxFiles point to a remote repo. For correctness, I think this should be `hc.Repo != "" || hc.IsFluxFile()`

You'd then have to change your `if/else` logic below a bit

Reply:
Yes, I think this is not the right abstraction... will try to make it better